arena/arena_mqtt.py

```diff
@@ -244,15 +222,6 @@ def network_latency_update(self):
         # publish empty message with QoS of 2 to update latency
         self.mqttc.publish(self.latency_topic, "", qos=2)
 
-    def msg_rate_update(self):
-        """Update Message rate"""
-        elapsed = datetime.now() - self.msg_rate_time_start
-        if elapsed.seconds > 0:
-            self.msg_io['rcv_msgs_per_sec'] = round(self.msg_io['rcv_msgs']  / elapsed.seconds, 2)
-            self.msg_io['pub_msgs_per_sec'] = round(self.msg_io['pub_msgs']  / elapsed.seconds, 2)
-            self.msg_io['rcv_queue_size'] = self.msg_queue.qsize()
-            self.msg_io['pub_queue_size'] = len(self.mqttc._out_packet)
```

Comment:
Do you want to keep these two queue metrics (in and out)?

Reply:
I moved these metrics into utils/program_info.py. At startup, it creates a periodic task to update these based on send/receive stats:

```
    def _update_stats(self):
        """Update stats (called by periodic task); Execute callback if defined """
        elapsed = datetime.now() - self._msg_rate_time_start
        if elapsed.seconds > 0:
            self.rcv_msgs_per_sec = round(self.rcv_msgs  / elapsed.seconds, 2)
            self.pub_msgs_per_sec = round(self.pub_msgs  / elapsed.seconds, 2)
        if self._update_callback: self._update_callback(self)
```

The scene (scene.py) overloads on_message and on_publish which updates program_info stats (by calling run_info.msg_rcv() and run_info.msg_publish).

Reply:
I will see how to add queue sizes back into that.